src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -54,14 +65,140 @@ public Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext,
+        {
+            return GetVariantAsync<T, TContext>(feature, appContext, true, cancellationToken);
+        }
+
+        public ValueTask<T> GetVariantAsync<T>(string feature, CancellationToken cancellationToken)
+        {
+            return GetVariantAsync<T, object>(feature, null, false, cancellationToken);
+        }
+
+        private async ValueTask<T> GetVariantAsync<T, TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
+        {
+            if (feature == null)
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            FeatureVariant variant = null;
+
+            FeatureDefinition featureDefinition = await _featureDefinitionProvider
+                .GetFeatureDefinitionAsync(feature, cancellationToken)
+                .ConfigureAwait(false);
+
+            if (featureDefinition == null)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.MissingFeatureFilter,
+                    $"The feature declaration for the dynamic feature '{feature}' was not found.");
+            }
+
+            if (string.IsNullOrEmpty(featureDefinition.Assigner))
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.InvalidConfiguration,
+                    $"Missing feature variant assigner name for the feature {feature}");
+            }
+
+            if (featureDefinition.Variants == null)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.InvalidConfiguration,
+                    $"No variants are registered for the feature {feature}");
+            }
+
+            FeatureVariant defaultVariant = null;
+
+            foreach (FeatureVariant v in featureDefinition.Variants)
+            {
+                if (v.Default)
+                {
+                    if (defaultVariant != null)
+                    {
+                        throw new FeatureManagementException(
+                            FeatureManagementError.InvalidConfiguration,
+                            $"Multiple default variants are registered for the feature '{feature}'.");
+                    }
+
+                    defaultVariant = v;
+                }
+
+                if (v.ConfigurationReference == null)
+                {
+                    throw new FeatureManagementException(
+                        FeatureManagementError.MissingConfigurationReference,
+                        $"The variant '{variant.Name}' for the feature '{feature}' does not have a configuration reference.");
+                }
+            }
+
+            if (defaultVariant == null)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.InvalidConfiguration,
```

Comment:
It's not clear to me when we will throw a generic InvalidConfiguration error and when we will throw a specific error, because we have have thrown a MissingDefaultVariant error here and an AmbiguousDefaultVariant a few lines above.

Reply:
Just a curious observer here, any chance this could be less likely to throw by taking in a default value in the code and logging warnings for example? The feature flagging lib throwing is scary if it cannot resolve the default value.